cmu_graphics/cmu_graphics.py

```diff
@@ -1221,12 +1222,12 @@ def setActiveScreen(screen, suppressEvent=False):
     redrawAllFnName = f'{screen}_redrawAll'
     if redrawAllFnName not in app._app.userGlobals:
         raise Exception(f'Screen {screen} requires {redrawAllFnName}()')
-    app._app.activeScreen = screen
-    if not suppressEvent:
-        app._app.callUserFn(f'{screen}_onScreenActivate', ())
+    pygame.event.post(pygame.event.Event(app.SET_ACTIVE_SCREEN, newScreen=screen))
 
 
 def runAppWithScreens(initialScreen, *args, **kwargs):
+    app.SET_ACTIVE_SCREEN = pygame.event.custom_type()
```

Comment:
Cool! This is better than using math on the `USEREVENT`. This should be a global at the top of the file though :-)